src/storage/query/QueryBaseProcessor.inl

```diff
@@ -522,9 +597,16 @@ kvstore::ResultCode QueryBaseProcessor<REQ, RESP>::collectEdgeProps(
                     continue;
                 }
             }
+
+            // Check if the schema has TTL
+            if (!checkDataExpiredForTTL(schema.get(), reader.get())) {
```

Comment:
I suggest we check ttl before checking the expression logic.

Reply:
Good catch!